pkg/providers/vsphere/setupuser/setupuser.go

```diff
@@ -0,0 +1,185 @@
+package setupuser
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	vSphereRootPath = "/"
+)
+
+// GovcClient specifies govc functions required to configure a vsphere user.
+type GovcClient interface {
+	CreateUser(ctx context.Context, username string, password string) error
+	UserExists(ctx context.Context, username string) (bool, error)
+	CreateGroup(ctx context.Context, name string) error
+	GroupExists(ctx context.Context, name string) (bool, error)
+	AddUserToGroup(ctx context.Context, name string, username string) error
+	RoleExists(ctx context.Context, name string) (bool, error)
+	CreateRole(ctx context.Context, name string, privileges []string) error
+	SetGroupRoleOnObject(ctx context.Context, principal string, role string, object string, domain string) error
+}
+
+// SetupGOVCEnv creates appropriate govc environment variables to build govc client.
+func SetupGOVCEnv(ctx context.Context, vsuc *VSphereSetupUserConfig) error {
+	os.Setenv("GOVC_URL", vsuc.Spec.Connection.Server)
+	os.Setenv("GOVC_INSECURE", strconv.FormatBool(vsuc.Spec.Connection.Insecure))
```

Comment:
Let's test and double-check if we can avoid using GOVC_INSECURE by specifying thumbprint for self-signed certs and also see if it can be avoided in trusted cert cases.

Reply:
Breaking out this work into a future ticket based on offline conversation.
Ticket: https://github.com/aws/eks-anywhere/issues/3708